src/specs/networking.md

```diff
@@ -66,3 +71,40 @@ Accepting a `WireTxPayForMessage` into the mempool requires different logic than
+| `share`    | [Share](./data_structures.md#share)                                                         | The share.                                                                                        |
+| `proof`    | [NamespaceMerkleTreeInclusionProof](./data_structures.md#namespacemerkletreeinclusionproof) | The Merkle proof of the share in the offending row or column root.                                |
+| `isCol`    | `bool`                                                                                      | A Boolean indicating if the proof is from a row root or column root; `false` if it is a row root. |
+| `position` | `uint64`                                                                                    | The index of the share in the offending row or column.                                            |
+
+
+### BadEncodingFraudProof
+
+| name          | type                                                                                  | description                                                                                           |
+|---------------|---------------------------------------------------------------------------------------|------------------------------------------------------------------------------------------------------|
+| `height`      | [Height](./data_structures.md#type-aliases)                                           | Height of the block with the offending row or column.                                                 |
+| `shareProofs` | [ShareProof](#shareproof)`[]`                                                         | The available shares in the offending row or column.                                                 |
+| `isCol`       | `bool`                                                                                | A Boolean indicating if it is an offending row or column; `false` if it is a row.                     |
+| `position`    | `uint64`                                                                              | The index of the offending row or column in the square.                                               |
+## Invalid State Update
+
+If a malicious block producer incorrectly computes the state, a fraud proof for this can be presented. We assume that the light clients have the [AvailableDataHeader](./data_structures.md#availabledataheader) and the [Header](./data_structures.md#header) for each block. Hence, given a [ShareProof](#shareproof), they can verify if the `rowRoot` or `colRoot` specified by `isCol` and `position` commits to the corresponding [Share](./data_structures.md#share). Similarly, given the `height` of a block, they can access all elements within the [AvailableDataHeader](./data_structures.md#availabledataheader) and the [Header](./data_structures.md#header) of the block. 
```

Comment:
We can't assume that for super-light clients -- do we want to allow them to receive state fraud proofs?

Reply:
I agree that we cannot assume this for super-light clients but they are secure only under an honest majority even with the ability to receive fraud proofs according to the [spec](https://github.com/celestiaorg/celestia-specs/blob/master/src/specs/node_types.md). Thus, fraud proofs do not seem to contribute to their security assumption as much as it does for regular light clients.

Reply:
To expand on this, the reasoning is that a majority of validators can trick superlight clients by withholding data, and thus they'll never see a state transition or bad transition fraud proof. In fact, the requirement of superlight clients to receive fraud proofs should be removed, as it may place unnecessary burden on the implementation.